include/s3select.h

```diff
@@ -606,8 +612,10 @@ struct s3select : public bsc::grammar<s3select>
       between_predicate = (arithmetic_expression >> bsc::str_p("between") >> arithmetic_expression >> bsc::str_p("and") >> arithmetic_expression)[BOOST_BIND_ACTION(push_between_filter)];
 
       in_predicate = (arithmetic_expression >> bsc::str_p("in") >> '(' >> arithmetic_expression[BOOST_BIND_ACTION(push_in_predicate_first_arg)] >> *(',' >> arithmetic_expression[BOOST_BIND_ACTION(push_in_predicate_arguments)]) >> ')')[BOOST_BIND_ACTION(push_in_predicate)];
+      
+      like_predicate_no_escape = (arithmetic_expression >> bsc::str_p("like") >> arithmetic_expression)[BOOST_BIND_ACTION(push_like_predicate)];
 
-      like_predicate = (arithmetic_expression >> bsc::str_p("like") >> arithmetic_expression)[BOOST_BIND_ACTION(push_like_predicate)];
+      like_predicate_escape = (arithmetic_expression >> bsc::str_p("like") >> arithmetic_expression >> bsc::str_p("escape") >> arithmetic_expression)[BOOST_BIND_ACTION(push_like_predicate_escape)];
```

Comment:
incase specs , define only constant string (is it?)
it's better to use the string-rule instead of expression-rule, in that way the syntax definition ("stronger") will filter out wrong queries.